docs/configuration/configuration.md

```diff
@@ -2145,3 +2186,29 @@ tls_config:
 There is a list of
 [integrations](https://prometheus.io/docs/operating/integrations/#remote-endpoints-and-storage)
 with this feature.
+
+### `oauth2`
+
+OAuth 2.0 authentication using the client credentials grant type.
+Prometheus fetches an access token from the specified endpoint with 
+the given client access and secret keys.
+
+```yaml
+[ client_id: <string> ]
```

Comment:
I am wondering now if we should make some of them mandatory. What does the spec say?

Reply:
Definitely mandatory. I think my original code checked for that but it got lost along the way, my bad.

Reply:
I can go deal with it later.

Reply:
prometheus/common#294

Reply:
I have release 2.23.0, you can update and remove the square brackets :)